StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1533,11 +1547,13 @@ public class STPPaymentHandler: NSObject {
                         if
                             let paymentMethod = paymentIntent.paymentMethod,
                             !STPPaymentHandler._isProcessingIntentSuccess(for: paymentMethod.type),
-                            paymentIntent.status == .processing && retryCount > 0
+                            paymentIntent.status == .processing,
+                            pollingBudget?.canPoll ?? true
                         {
-                            self._retryAfterDelay(retryCount: retryCount) {
+                            let processingPollingBudget = pollingBudget ?? PollingBudget(startDate: startDate, duration: 1)
```

Comment:
Looking closer, we shouldn't use a 1 second timeout here. This is a different case than the LPM-cancel problem we're solving. In this case we're **sure** that the PaymentIntent will transition. The "polling budget" should be long, like 30 seconds.